### PR TITLE
[refactor embeddings]pixart-alpha

### PR DESCRIPTION
**Pull request by yiyixuxu**
part of my embedding refactor, separated by model/pipeline so it is easier to work with 

this PR focuses on embeddings only used in Pixar-alpha:  i.e. `CombinedTimestepSizeEmbeddings` and `CaptionProjection`:
1. rename them to `PixArtAlphaCombinedTimestepSizeEmbeddings` and `PixArtAlphaTextProjection` so it is clear that these embeddings are only used in PixArt-Alpha
2. remove code that's not needed (let me know if I got anything wrong here)

### CODE REVIEW

File: src/diffusers/models/embeddings.py
Comment (patrickvonplaten) [inline src/diffusers/models/embeddings.py:758]: very nice refactor!

**Review by patrickvonplaten — APPROVED** [review]

**Review by sayakpaul — COMMENTED** [review]
LGTM if the SLOW tests pass. Could you please run the slow tests with these changes as well?

File: src/diffusers/pipelines/pixart_alpha/pipeline_pixart_alpha.py
Comment (sayakpaul) [inline src/diffusers/pipelines/pixart_alpha/pipeline_pixart_alpha.py:859]: Seems like a new addition?

Comment (yiyixuxu) [reply src/diffusers/pipelines/pixart_alpha/pipeline_pixart_alpha.py:859]: not really - it gets duplicated later inside `CombinedTimestepSizeEmbeddings`

https://github.com/huggingface/diffusers/blob/6976cab7caa959bff46896c52ec03d9d414fc689/src/diffusers/models/embeddings.py#L758

Comment (sayakpaul) [reply src/diffusers/pipelines/pixart_alpha/pipeline_pixart_alpha.py:859]: Very nice <3

Comment (yiyixuxu) [reply src/diffusers/pipelines/pixart_alpha/pipeline_pixart_alpha.py:859]: @sayakpaul 
fast tests fail because there are some randomly initialized weights in some components. I think we need to put `torch.manual_seed(0)` before making each component e.g.
https://github.com/huggingface/diffusers/blob/6976cab7caa959bff46896c52ec03d9d414fc689/tests/pipelines/pixart/test_pixart.py#L67

should we open a new PR to only update the tests, and I rebase after that? I'm not comfortable updating tests directly from this PR since I updated the code

Comment (sayakpaul) [reply src/diffusers/pipelines/pixart_alpha/pipeline_pixart_alpha.py:859]: But that wasn't the case before. Wonder what changed.

File: src/diffusers/models/transformer_2d.py
Comment (DN6) [inline src/diffusers/models/transformer_2d.py:238]: Might actually be worth breaking up Transformer2D up into a dedicated one for PixArt.

Comment (sayakpaul) [reply src/diffusers/models/transformer_2d.py:238]: For a future PR, yeah? I am happy to work on it once this is merged.

Comment (yiyixuxu) [reply src/diffusers/models/transformer_2d.py:238]: Definitely for a future PR

But I think we should refactor transformers and UNet after we clean up all the lower-level classes and make such decisions all together so it will be consistent

Comment (sayakpaul) [reply src/diffusers/models/transformer_2d.py:238]: Yes 100 percent!

**Pull request merged**
